service/instance.py

```diff
@@ -442,11 +442,21 @@ def redeploy_instance(
 
     if type(esh_driver) == AtmosphereMockDriver:
         return
+    # Force a metadata update to avoid reverting to previous state.
```

Comment:
On first blush this doesn't seem right. get_idempotent_deploy_chain works by looking at the current state, and acting. So no matter what state the instance is in (before your call here to set "initializing"), get_idempotent_deploy should be able to act accordingly right? 
If you look at this code, you'd ask yourself, "why would we force to avoid reverting?", because its not obvious within the context of *this* function. This change only makes sense when you know how another chunk of the code works in `get_deployment_chain`, which is a hint that code is being changed in the wrong place.

Reply:
You are probably right, that this change is happening in the wrong place.

The "Problem scenario" is:
- GUI (via user click) requests `reboot/hard reboot/redeploy`
- API forwards request to cloud, which accepts
- GUI begins Instance Polling
- API returns the last known instance state (In some cases, prior to the change being executed)
- GUI 'reverts' state back to a "known final state" (active, deploy_error, etc.)
- GUI never polls again, so user thinks this is the "end of the line"

- On page refresh//next poll, the action will have happened, and the appropriate status will be updated.

This is an *improvement* on the current behavior, in that it will avoid the 'final state, stop polling' step.. But it is not a *fix* for the larger bug described.. (I would be fine with writing up a JIRA to be addressed later, though..)